### PR TITLE
[DDW-496] Scrollable dialogs improvement

### PR DESCRIPTION
**Pull request by yakovkaravelov**
This PR improved scrollable dialogs layout
[Jira Ticket](https://jira.iohk.io/browse/DDW-496)

## Todos
- [x] Improve scrollable dialogs layout

## Issues tested on build 15751 
- [x] Is stakepools dialog the only one changed and tested? Here for the add dialog, [we can see that](https://user-images.githubusercontent.com/2539345/102886753-bccf7e80-4466-11eb-9f88-ad448512b53c.png) there is no more top distance between title and dialog-header. Also when scrolling we can see that [top area is not per design](https://user-images.githubusercontent.com/2539345/102886918-0c15af00-4467-11eb-9c58-a9df95fef4dc.png).
- [x] Restore dialog has ["steps" part](https://user-images.githubusercontent.com/2539345/102887015-37000300-4467-11eb-99c3-fcef7dfa4995.png) scrollable.
- [x] More wrong header size examples: [here](https://user-images.githubusercontent.com/2539345/102887134-6adb2880-4467-11eb-9a48-c8e905528e60.png), [here](https://user-images.githubusercontent.com/2539345/102887360-d8875480-4467-11eb-95b2-6d070a9fbb76.png)
- [x] End of scrollbar [doesn't match](https://user-images.githubusercontent.com/2539345/102896563-a8e04880-4477-11eb-8abe-5a3c3985f920.png) end of content 

 

## Testing Checklist
- [Slack QA thread](https://input-output-rnd.slack.com/archives/GGKFXSKC6/p1608636165081400)

### App
- [x] Check all the scrollable dialogs and make sure they are matching with wireframe. https://zpl.io/VkABvQg

## Test Cases

**Scenario 1: Validate the footer area when scrolling**
```Gherkin
Given that I am on a [Dialog]
And I can see the scrollbar on the right
When I scroll down
Then the bottom area is fixed to the bottom
```

**Scenario 2: Validate the header area when scrolling**
```Gherkin
Given that I am on a [Dialog]
And I can see the scrollbar on the right
When I scroll up
Then the header area is fixed to the top
```

**Scenario 3: Validate the middle area when scrolling**
```Gherkin
Given that I am on a [Dialog]
And I can see the scrollbar on the right
When I scroll up and down
Then only the area between the header and the bottom scrolls accordingly
```

**Scenario 4: Validate non-scrollable dialogs**
```Gherkin
Given that I am on a [Dialog]
And I can't see the scrollbar on the right
When I validate the space between the header and the content
Then I can see there are 20px between the title and the content
And 10 px between the button on the footer and the content
```

**Validate these dialogs:**
- Delegate wallet
- Create a new wallet
- Restore a wallet 
- Pair Hardware Wallet
- Wallet recovery phrase verification
- Delete Wallet
- Change password
- Share wallet address

## Testing Summary

English
image
image

Japanese
image
image

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

**Review by DeeJayElly — APPROVED** [review]
Looks ok @yakovkaravelov

**Review by a-rukin — CHANGES_REQUESTED** [review]
Fix issues in PR description

**Comment by mioriohk** [comment]
Looks OK to me on build 15751. Need to double check with Yakov which dialogs are exactly scrollable and which aren't.

**Comment by a-rukin** [comment]
> Looks OK to me on build 15751. Need to double check with Yakov which dialogs are exactly scrollable and which aren't.

This rule should be applied to all dialogs. As they can eventually become scrollable with a bigger content. 
This is a feature of dialog as a component, not just a fix for a spot with wrong padding...

**Comment by mioriohk** [comment]
> > Looks OK to me on build 15751. Need to double check with Yakov which dialogs are exactly scrollable and which aren't.
> 
> This rule should be applied to all dialogs. As they can eventually become scrollable with a bigger content.
> This is a feature of dialog as a component, not just a fix for a spot with wrong padding...

Thanks for the clarification @a-rukin . Apart from the design issues that you have found, looks OK to me so far. I am going to do another round of testing on this.

**Review by DeeJayElly — APPROVED** [review]
Looks ok @yakovkaravelov gj !!!

**Review by gnpf — APPROVED** [review]

**Comment by nikolaglumac** [comment]
@gnpf what about these? 
Screenshot 2021-01-08 at 18 04 28

**Comment by nikolaglumac** [comment]
We will not be merging this one without @a-rukin's approval.

**Comment by gnpf** [comment]
@nikolaglumac @yakovkaravelov The issues mentioned in the description are fixed, except this:

>  Restore dialog has "steps" part scrollable.

Check [this screenshot.](https://drive.google.com/file/d/1Eagc7_rwXZ4OIAWV55hUAIXJkrbuWx2h/view?usp=sharing)

**Comment by a-rukin** [comment]
@nikolaglumac @gnpf 
I'm sorry to say, but I can reproduce every issue from my list. 😢

**Comment by nikolaglumac** [comment]
@yakovkaravelov please get on a call with @a-rukin and fix the remaining issues. Thanks!

**Comment by mioriohk** [comment]
@yakovkaravelov Apart from what @a-rukin has found, here are the findings for build 15824:

Test case | Status | Comments
-- | -- | --
Scenario 2 | Failed | I could replicate the issue found by @gnpf [here](https://github.com/input-output-hk/daedalus/pull/2285#issuecomment-756915953). See [video here](https://drive.google.com/file/d/1rlmLWLj-64qkjvvRa-ENgwQm5UOxsNxT/view?usp=sharing)

**Comment by yakovkaravelov** [comment]
@a-rukin @nikolaglumac Fixed the issues found by Sasha.
Please review again.
Thanks.

**Comment by a-rukin** [comment]
@yakovkaravelov excellent!!
One small thing I found is that scrollbar [lost it's rounded corners](https://user-images.githubusercontent.com/2539345/104566934-48e85680-565f-11eb-92f6-b9b472014cb3.png)

File: source/renderer/app/components/widgets/Dialog.scss [hunk 1]
```diff
@@ -26,12 +26,22 @@
...
     scroll-behavior: smooth;
 
+    &::-webkit-scrollbar {
+      width: 4px;
+    }
+
+    &::-webkit-scrollbar-thumb:vertical {
+      border: 0;
+      border-bottom: 20px solid transparent;
+      border-radius: 0;
```
Comment (a-rukin) [inline source/renderer/app/components/widgets/Dialog.scss:42]: @yakovkaravelov think this one is redundant

**Review by a-rukin — CHANGES_REQUESTED** [review]

**Review by mioriohk — APPROVED** [review]
LGTM on build 15882. 👍

**Review by nikolaglumac — APPROVED** [review]
Great work @yakovkaravelov 👍

**Review by a-rukin — APPROVED** [review]
Great! Perfect! 👍

**Pull request merged**